pkg/cnbutils/privacy/privacy.go

```diff
@@ -0,0 +1,47 @@
+package privacy
+
+import (
+	"strings"
+
+	containerName "github.com/google/go-containerregistry/pkg/name"
+)
+
+func FilterBuildpacks(buildpacks []string) []string {
+	result := make([]string, 0, len(buildpacks))
+	for _, buildpack := range buildpacks {
+		ref, err := containerName.ParseReference(strings.ToLower(buildpack))
+		if err != nil {
+			result = append(result, "<error>")
+			continue
+		}
+
+		registry := ref.Context().Registry.Name()
+		repository := ref.Context().RepositoryStr()
+
+		if registry == "gcr.io" && strings.HasPrefix(repository, "paketo-buildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "index.docker.io" && strings.HasPrefix(repository, "paketobuildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "gcr.io" && strings.HasPrefix(repository, "buildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "public.ecr.aws" && strings.HasPrefix(repository, "heroku-buildpacks/") {
+			result = append(result, buildpack)
+		} else {
+			result = append(result, "<retracted>")
+		}
+	}
+	return result
+}
+
+var allowedEnvKeys = map[string]interface{}{"BP_JVM_VERSION": nil, "BP_NODE_VERSION": nil}
```

Comment:
I would assume, that we have more well-known environment variables that we want to put here, no?

Reply:
I looked at the most relevant buildpacks (for us, e.g. ca-certificates, jvms, tomcat, maven, gradle, node-engine, npm, yarn) and added `BP_JVM_TYPE` (`JDK` or `JRE`) and `BP_TOMCAT_VERSION` (the other tools like maven don't allow selecting the version).

Many of the ones I did not include contain paths, which could leak some information about the user.
Others (like the JMX port) should be only relevant at runtime.

We can expand this list any time, I think it's a good first version now.